src/Faker/Core/Number.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Faker\Core;
+
+use Faker\Extension;
+
+/**
+ * @experimental This class is experimental and does not fall under our BC promise
+ */
+final class Number implements Extension\NumberExtension
+{
+    public function numberBetween(int $min = 0, int $max = 2147483647): int
+    {
+        $int1 = $min < $max ? $min : $max;
+        $int2 = $min < $max ? $max : $min;
+
+        return mt_rand($int1, $int2);
+    }
+
+    public function randomDigit(): int
+    {
+        return mt_rand(0, 9);
+    }
+
+    public function randomDigitNot(int $except): int
+    {
+        $result = self::numberBetween(0, 8);
+
+        if ($result >= $except) {
+            ++$result;
+        }
+
+        return $result;
+    }
+
+    public function randomDigitNotNull(): int
```

Comment:
This is a weird method name where i guess whereNotZero is meant?

Since the return type is never null

Reply:
You're right @pimjansen, I renamed the method.